m4/ax_compare_version.m4

```diff
@@ -0,0 +1,177 @@
+# ===========================================================================
+#    https://www.gnu.org/software/autoconf-archive/ax_compare_version.html
+# ===========================================================================
```

Comment:
Why did you copy `ax_compare_version.m4` and `ax_compiler_vendor.m4` into our repository?
`autoconf-archive` is our hard dependency. Are these macros outdated in distro's official packages?

Reply:
I wanted to be sure these macros will be available independently from autoconf-archive version. Though most probably you are right, these are very basic macros, this has low risk. Let me check if we can compile without them.

Reply:
(I've only checked with latest package in Xenial.)